var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -47,6 +48,7 @@ class Netcdf(Package):
 
     # Required for NetCDF-4 support
     depends_on("zlib")
+    depends_on("hdf5@:1.8.999", when='+hdf5_18')
```

Comment:
`hdf5@:1.8` does indeed work, you can test this yourself by adding it to the NetCDF package, removing the HDF5 preferred version, and running `spack spec netcdf`.